rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -85,9 +89,39 @@ rmw_create_client(
     }
   }
 
-  CustomClientInfo * info = nullptr;
+  // If FASTRTPS_DEFAULT_PROFILES_FILE defined, fill subscriber attributes with a subscriber
+  // profile profile located based of topic name defined by _create_topic_name(). If no profile is
+  // found, a search with profile_name "service"
+  // is attempted. Else, use the default attributes.
+  std::string topic_name_fallback = "client";
   eprosima::fastrtps::SubscriberAttributes subscriberParam;
+  eprosima::fastrtps::fixed_string<255> sub_topic_name = _create_topic_name(
+    qos_policies, ros_service_response_prefix, service_name, "Reply");
+  Domain::getDefaultSubscriberAttributes(subscriberParam);
+  if (std::getenv("FASTRTPS_DEFAULT_PROFILES_FILE") != nullptr) {
```

Comment:
`std::getenv` triggered warnings on Windows. Compiler suggests using `_dupenv_s` instead. Same for below.

Reply:
This should probably use the platform independent function `rcutils_get_env()`.